apps/api/src/app/controls/usecases/get-control-schema/get-control-schema.usecase.ts

```diff
@@ -0,0 +1,27 @@
+import { Injectable } from '@nestjs/common';
+import { ActionStepEnum, actionStepSchemas, ChannelStepEnum, channelStepSchemas, StepType } from '@novu/framework';
+
+import { GetControlSchemaCommand } from './get-control-schema.command';
+import { StepOutputSchema } from '../../types';
+
+@Injectable()
+export class GetControlSchema {
+  async execute(command: GetControlSchemaCommand): Promise<StepOutputSchema> {
+    return getStepOutput(command.stepType);
+  }
+}
+
+export const getStepOutput = (stepType: StepType): StepOutputSchema => {
+  return StepTypeToOutputMap[stepType];
+};
+
+export const StepTypeToOutputMap = {
+  [ChannelStepEnum.SMS]: channelStepSchemas[ChannelStepEnum.SMS].output,
+  [ChannelStepEnum.EMAIL]: channelStepSchemas[ChannelStepEnum.EMAIL].output,
+  [ChannelStepEnum.PUSH]: channelStepSchemas[ChannelStepEnum.PUSH].output,
+  [ChannelStepEnum.CHAT]: channelStepSchemas[ChannelStepEnum.CHAT].output,
+  [ChannelStepEnum.IN_APP]: channelStepSchemas[ChannelStepEnum.IN_APP].output,
+  [ActionStepEnum.DELAY]: actionStepSchemas[ActionStepEnum.DELAY].output,
+  [ActionStepEnum.DIGEST]: actionStepSchemas[ActionStepEnum.DIGEST].output,
+  [ActionStepEnum.CUSTOM]: {},
```

Comment:
It might be worthwhile to use an empty object schema that doesn't allow any properties for the custom step to ensure that if custom is erroneously used in the client, it can be handled correctly by the form renderers.
```suggestion
  [ActionStepEnum.CUSTOM]: {
    type: 'object',
    properties: {},
    required: [],
    additionalProperties: false,
  },
```

Reply:
Good point thanks